src/Tribe/RSVP.php

```diff
@@ -580,20 +909,30 @@ public function generate_tickets() {
 
 		// If there are details missing, we return to the event page with the rsvp_error
 		if ( false === $attendee_details ) {
-			$url = get_permalink();
-			$url = add_query_arg( 'rsvp_error', 1, $url );
-			wp_redirect( esc_url_raw( $url ) );
-			tribe_exit();
+			if ( $redirect ) {
+				$url = get_permalink();
+				$url = add_query_arg( 'rsvp_error', 1, $url );
+				wp_redirect( esc_url_raw( $url ) );
```

Comment:
* :warning: wp_redirect() found. Using wp_safe_redirect(), along with the allowed_redirect_hosts filter if needed, can help avoid any chances of malicious redirects within code. It is also important to remember to call exit() after a redirect so that no other unwanted code is executed.
* :no_entry_sign: `wp_redirect()` should almost always be followed by a call to `exit;`.

<sub>hash: 37372fa7e5e5b40a56e168fa665276e9</sub>